stdlib/format.ml

```diff
@@ -80,6 +83,12 @@ type pp_token =
   | Pp_newline                 (* to force a newline inside a box *)
   | Pp_if_newline              (* to do something only if this very
                                   line has been broken *)
+  | Pp_break_or_text_if_newline of fits_or_breaks option * string
```

Comment:
I suggested this generalization with the idea that we could fine uses for the before/after fields in the future (not unlike what you are doing with `Pp_break`), so having the more expressive instruction in the abstract machine would make it easier to generalize. (And it actually makes the implementation nicer, because the interpreter is simpler.)